File: frontend/src/routes/Offer.js
import { useParams } from 'react-router-dom';
import React from 'react';
import './../styles/routes/offer-style.css';
import Appartment from './../assets/images/appartment1.jpg';

export default function Offer() {
    const { id } = useParams();
    const [offerData, setOfferData] = React.useState({}); 

    React.useEffect(() => {
        fetch(`http://localhost:9090/bff/offers/${id}`)
            .then(res => res.json())
            .then(data => setOfferData(data))
            .catch(error => console.log('Error during fetching offer data:', error));
    }, [id]);

    return(
        <div className="offerContainer">
            <img src={Appartment} alt='appartment' className='appartmentImage'/> 
            <div className='title'>{offerData.offerName}</div>
            <div className='price'>Cena całkowita: {offerData.fullPrice} pln / miesiąc</div>
            <div className='address'>{offerData.city}, {offerData.street}, {offerData.district}, {offerData.zip}</div>
            <div className='additionalInfo'>Rok budowy: {offerData.buildingYear}, Powierzchnia: {offerData.area}, Liczba pokoi: {offerData.roomsNumber},
            Piętro: {offerData.estateLevel}, Aneks kuchenny: {offerData.annexKitchen}, Winda: {offerData.elevator}, Zwierzęta mile widziane: {offerData.animals}</div>
        </div>
    )
}

File: client/src/main/java/com/kierznowski/rentalapp/client/model/Offer.java
package com.kierznowski.rentalapp.client.model;

import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

@Data
@RequiredArgsConstructor
//@NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
public class Offer {

    private Long id;
    private UserDTO user;
    private Date createdAt;
    private String offerName;
    private String city;
    private String district;
    private String street;
    private String zip;
    private int buildingYear;
    private BigDecimal fullPrice;
    private BigDecimal basePrice;
    private BigDecimal additionalPrice;
    private int area;
    private int roomsNumber;
    private int estateLevel;
    private boolean garage;
    private boolean annexKitchen;
    private boolean elevator;
    private boolean animals;
    private List<Long> imageIds;
}
